libcudacxx/include/cuda/std/__bit/reference.h

```diff
@@ -77,68 +87,84 @@ class __bit_reference
     return *this;
   }
 
-  _LIBCUDACXX_INLINE_VISIBILITY __bit_reference& operator=(const __bit_reference& __x) noexcept
+#if _CCCL_STD_VER >= 2023
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY constexpr const __bit_reference&
+  operator=(bool __x) const noexcept
+  {
+    if (__x)
+    {
+      *__seg_ |= __mask_;
+    }
+    else
+    {
+      *__seg_ &= ~__mask_;
+    }
+    return *this;
+  }
+#endif // C++23+
+
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 __bit_reference&
+  operator=(const __bit_reference& __x) noexcept
   {
     return operator=(static_cast<bool>(__x));
   }
 
-  _LIBCUDACXX_INLINE_VISIBILITY void flip() noexcept
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 void flip() noexcept
   {
     *__seg_ ^= __mask_;
   }
-  _LIBCUDACXX_INLINE_VISIBILITY __bit_iterator<_Cp, false> operator&() const noexcept
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 __bit_iterator<_Cp, false>
+  operator&() const noexcept
+  {
+    return __bit_iterator<_Cp, false>(__seg_, static_cast<unsigned>(_CUDA_VSTD::__libcpp_ctz(__mask_)));
+  }
+
+  friend inline _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 void
+  swap(__bit_reference<_Cp> __x, __bit_reference<_Cp> __y) noexcept
```

Comment:
We should be able to drop the template argument from `__bit_reference`. Also should this take references?
```suggestion
  swap(__bit_reference& __x, __bit_reference& __y) noexcept
```

Applies below

Reply:
May do cleanup of the template arguments later, but since this is a proxy type, passing them by value will actually make the compiler's job easier at generating the optimal code.